sonar-plugin-api/src/main/java/org/sonar/api/utils/DateUtils.java

```diff
@@ -19,30 +19,44 @@
  */
 package org.sonar.api.utils;
 
+import java.time.DateTimeException;
 import java.time.Instant;
 import java.time.LocalDate;
+import java.time.LocalDateTime;
 import java.time.OffsetDateTime;
 import java.time.ZoneId;
 import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeFormatterBuilder;
 import java.time.format.DateTimeParseException;
+import java.time.temporal.ChronoField;
 import java.time.temporal.ChronoUnit;
+import java.time.temporal.TemporalAccessor;
 import java.util.Date;
 import javax.annotation.CheckForNull;
 import javax.annotation.Nullable;
 
 import static com.google.common.base.Preconditions.checkArgument;
+import static java.time.format.DateTimeFormatter.ISO_DATE;
+import static java.time.format.DateTimeFormatter.ISO_DATE_TIME;
 
 /**
- * Parses and formats <a href="https://docs.oracle.com/javase/8/docs/api/java/text/SimpleDateFormat.html#rfc822timezone">RFC 822</a> dates.
+ * Parses and formats <a href="https://docs.oracle.com/javase/8/docs/api/java/time/format/DateTimeFormatter.html#ISO_DATE_TIME">ISO-8601</a> dates and date times.
  * This class is thread-safe.
  *
  * @since 2.7
  */
 public final class DateUtils {
+  /**
+   * @deprecated since 7.0
+   */
+  @Deprecated
   public static final String DATE_FORMAT = "yyyy-MM-dd";
+  /**
+   * @deprecated since 7.0
+   */
+  @Deprecated
   public static final String DATETIME_FORMAT = "yyyy-MM-dd'T'HH:mm:ssZ";
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Do not forget to remove this deprecated code someday. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133)